test/integration/kubernetes_deploy_test.rb

```diff
@@ -1006,4 +1007,32 @@ def test_raise_on_yaml_missing_kind
       "  datapoint: value1"
     ], in_order: true)
   end
+
+  def test_crd_can_be_successful
+    assert_deploy_success(deploy_fixtures("hello-cloud", subset: ["crd.yml"]))
+    assert_logs_match_all([
+      "Phase 1: Initializing deploy",
+      "Detected non-namespaced resouce which will never be pruned:",
+      " - CustomResourceDefinition/mails.stable.example.io",
+      "Phase 2: Checking initial resource statuses",
+      "Deploying CustomResourceDefinition/mails.stable.example.io (timeout: 30s)",
+      %r{CustomResourceDefinition/mails.stable.example.io\s+Succeeded}
+    ])
+  end
+
+  def test_crd_can_fail
+    assert_deploy_success(deploy_fixtures("hello-cloud", subset: ["crd.yml"]))
+    result = deploy_fixtures("hello-cloud", subset: ["crd.yml"]) do |f|
+      crd = f.dig("crd.yml", "CustomResourceDefinition").first
+      names = crd.dig("spec", "names")
+      crd["metadata"]["name"] = 'mis-matched.stable.example.io'
+      names["plural"] = 'mis-matched'
+    end
+    assert_deploy_failure(result)
+    assert_logs_match_all([
+      "Deploying CustomResourceDefinition/mails.stable.example.io (timeout: 30s)",
+      "CustomResourceDefinition/mis-matched.stable.example.io: FAILED",
+      "Final status: ListKindConflict"
```

Comment:
This is a little cryptic... Does the condition's message have additional information we're not printing?